recipes-devtools/sdk-files/files/environment-setup-template.tmpl

```diff
@@ -21,3 +21,9 @@ export CFLAGS="${TARGET_CFLAGS}"
 export CXXFLAGS="${TARGET_CXXFLAGS}"
 export CPPFLAGS="${TARGET_CXXFLAGS}"
 export LDFLAGS="${TARGET_LDFLAGS}"
+
+export ARCH="${DISTRO_ARCH}"
```

Comment:
I use qemu-amd64 as MACHINE. Then, I got following error.

```
build@d5b543c5d372:~/work/sdk-sample/hello-proc-module_6.1$ make
make -C /home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/kernel M=/home/build/work/sdk-sample/hello-proc-module_6.1
make[1]: Entering directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
Makefile:775: arch/amd64/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/amd64/Makefile'.  Stop.
make[1]: Leaving directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
make: *** [Makefile:11: all] Error 2
```

according to the error log,  "'arch/amd64/Makefile" is wrong architecture name. it should be x86.

```
build@d5b543c5d372:~/work/sdk-sample/hello-proc-module_6.1$  ARCH=x86  make
make -C /home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/kernel M=/home/build/work/sdk-sample/hello-proc-module_6.1
make[1]: Entering directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (Debian 12.2.0-14) 12.2.0
  You are using:           x86_64-linux-gnu-gcc-12.bin (Debian 12.2.0-14) 12.2.0
  CC [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.o
  MODPOST /home/build/work/sdk-sample/hello-proc-module_6.1/Module.symvers
  CC [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.mod.o
  LD [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.ko
make[1]: Leaving directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
```

So , ARCH variable should not use DISTRO_ARCH  directory.

Reply:
To fix this ARCH issue, I add kernel-arch.bbclass based on linux-custom.inc and modify sdk environment setup file to use this class file.